### PR TITLE
document local minikube. Fixes https://github.com/tilt-dev/tilt/issues/3193

### LINKED ISSUE #3193: Docs for Minikube + local registry (was: Allow tilt to copy image from host)
Hi,

I do not know if tilt should do it or not (I let you decide), but due to my connexion I have an issue at first start of a cluster.
Sometime I delete my minikube cluster, and after creating a new one, the first launch is slow due to pulling all images.
I am wondering if we could have a new command to allow to copy a local image into the cluster so even if I delete the cluster I can copy/paste images and then run `tilt up` it will drastically increase the first launch.

What are your thoughts about it?

### PR DESCRIPTION
**Pull request by nicks**
Hello @maiamcc,

Please review the following commits I made in branch nicks/minikube:

f637541df50d13c64549d9262e52efaf7409c291 (2020-09-21 18:04:23 -0400)
document local minikube. Fixes https://github.com/tilt-dev/tilt/issues/3193

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics

### CODE REVIEW

**Comment by nicks** [comment]
a ton of this is just copied and pasted from the work we did on kind-local
https://github.com/tilt-dev/kind-local

i found that the minikube instructions recommended a bad approach, and the setup was very manual

**Comment by nicks** [comment]
unforuntately, had to back out the circleci test due to a minikube bug

**Comment by nicks** [comment]
there's some discussion here https://github.com/kubernetes/minikube/issues/7420, there might be a workaround, but going to do it on a separate pr

**Comment by maiamcc** [comment]
...whoops I made a buncha comments straight on [this commit](https://github.com/tilt-dev/minikube-local/commit/f637541df50d13c64549d9262e52efaf7409c291), transferring them over to this PR :-/

File: README.md [hunk 1]
```diff
@@ -1,2 +1,71 @@
 # minikube-local
+
 The best way to set minikube up for local development
+
+[![Build Status](https://circleci.com/gh/tilt-dev/minikube-local/tree/master.svg?style=shield)](https://circleci.com/gh/tilt-dev/minikube-local)
+
+When using Tilt with a [Minikube](https://minikube.sigs.k8s.io/docs/) cluster, 
+we recommend using a local registry for faster image pushing and pulling.
```
Comment (maiamcc) [inline README.md:8]: > faster image pushing and pulling.

seems like you joinked this from a doc where the tradeoff was local registry vs. remote registry -- the main Q for minikube is about local registry vs. hooking into minikube docker, I think the docs should discuss that instead

File: README.md [hunk 2]
```diff
@@ -1,2 +1,71 @@
...
+
+- Unlike with an in-cluster registry, you can reset the cluster without deleting the image cache.
+
+This makes it a great solution for iterative local development. But setting it up is awkward and fiddly. This script makes it easy.
+
+## How to Try It
+
+1) Install [Minikube](https://minikube.sigs.k8s.io/docs/)
+
+2) Copy the [minikube-with-registry.sh](minikube-with-registry.sh) somewhere on your path.
```
Comment (maiamcc) [inline README.md]:
```suggestion
2) Copy the [minikube-with-registry.sh](minikube-with-registry.sh) script somewhere on your path.
```

File: README.md [hunk 3]
```diff
@@ -1,2 +1,71 @@
...
+4) Try pushing an image.
+
+```
+docker tag alpine localhost:5000/alpine
+docker push localhost:5000/alpine
+```
+
+You can now use the image name `localhost:5000/alpine` in any resources you deploy to the cluster.
+
+[Tilt](https://tilt.dev) will automatically detect the local registry created by this script.
```
Comment (maiamcc) [inline README.md:52]: I thought Tilt's default was to hook into the minikube docker daemon -- if it detects a local registry, does that behavior override that behavior?

File: README.md [hunk 4]
```diff
@@ -1,2 +1,71 @@
...
+
+## How to Use it in CI
+
+We also have instructions for setting Minikube up with a local registry in
+
+- [.circleci](.circleci)
```
Comment (maiamcc) [inline README.md]: ...why is this a bullet point?

**Review by maiamcc — APPROVED** [review]
lgtm!

File: README.md (hunk 1 above)
Comment (nicks) [reply README.md:8]: i think you overestimate how well most minikube users are going to understand the minikube feature set. simplified the explanation here a bit

File: README.md (hunk 2 above)
Comment (nicks) [reply README.md]: done

File: README.md (hunk 3 above)
Comment (nicks) [reply README.md:52]: nope!

we should change tilt to prefer the local registry over the docker daemon.

HOWEVER! This script uses containerd, so Tilt will detect that there's no docker daemon connect to

File: README.md (hunk 4 above)
Comment (nicks) [reply README.md]: removed! this is vestigial from when i thought circleci would work :grimacing:

File: README.md (hunk 3 above)
Comment (maiamcc) [reply README.md:52]: Ooh clever! Nice. Can you open an issue/story for the local-registry-over-minikube-docker behavior?

Comment (nicks) [reply README.md:52]: done https://github.com/tilt-dev/tilt/issues/3795

**Pull request merged**